src/com.company/backtracking/WordBreak.java

```diff
@@ -0,0 +1,95 @@
+package backtracking;
+
+import java.util.*;
+
+public class WordBreak {
+    private Trie trie;
+    private boolean wordBreak(String str, List<String> dict) {
+        trie = new Trie();
+        for(String word : dict) {
+            trie.insertTrie(word);
+        }
+        return solveWordBreak(str, 0);
+    }
+
+    private boolean solveWordBreak(String str, int index) {
+        if(index >= str.length()) {
+            return true;
+        }
+        for(int i = index ; i < str.length(); i++) {
+            if(trie.searchTrie(str.substring(index, i+1))) {
+                return solveWordBreak(str, i + 1);
+            }
+        }
+        return false;
+    }
+
+    public static void main(String[] args) {
+        List<String> dict = new ArrayList<>(Arrays.asList("apple","pen"));
+        String word = "applepenapple";
+        System.out.println(new WordBreak().wordBreak(word, dict));
+    }
+
+
+
+    static class TrieNode {
```

Comment:
Why do we require a static class here?

Reply:
we can create static or non static inner class
This inner class will not be created for each instance of the outer class.It belongs to the class and each instance of the outer class share same static inner class